### ISSUE TITLE
En-passant flag inconsistency

### ISSUE THREAD

**Issue by yo35** [yo35/kokopu#27]
There is an inconsistent behavior regarding the en-passant flag between the checks performed in `Position.isLegal(..)` and the update strategy applied in `Position.play(..)`:
- In `Position.isLegal(..)`, en-passant set for a column implies that the position is compatible with a 2-square pawn move just before (i.e. for instance, if en-passant is set to e, and if Black is to play, it implies that e2 and e3 are empty, and that there is a white pawn in e4).
- In `Position.play(..)`, en-passant is set after a 2-square pawn move ONLY if the opponent has a pawn in position to achieve en-passant (i.e., in the previous example, only if there is a black pawn in d4 or f4).

This leads to unexpected behaviors, i.e. for instance:
```
var p1 = new kokopu.Position();
p1.play('e4');
var p2 = new kokopu.Position('rnbqkbnr/pppppppp/8/8/4P3/8/PPPP1PPP/RNBQKBNR b KQkq e3 0 1');
console.log(kokopu.Position.isEqual(p1, p2));
// return false
```

**Comment by yo35** [comment]
Fixed in 2.8.1.

As of now, `Position.play(..)` sets the en-passant flag after a 2-square pawn move, even if there is no opponent pawns on the squares from where en-passant capture could be initiated.

**Comment by mbutsykin** [comment]
@yo35 
What about i.e following case:

```
var kokopu = require("kokopu");

var p1 = new kokopu.Position();
p1.play('e4');
p1.play('Nf6');
p1.play('d4');

var p2 = new kokopu.Position();
p2.play('d4');
p2.play('NF6');
p2.play('e4');

kokopu.Position.isEqual(p1, p2);
// false
```

yes - en passant cells are different, but from game perspective - positions are equal.
i think it should work opposite - `kokopu` should ignore en-passant flag in case there are no pawns for it
for example here is [lichess position](https://lichess.org/analysis/fromPosition/rnbqkbnr/pppppppp/8/8/4P3/8/PPPP1PPP/RNBQKBNR_b_KQkq_e3_0_1#3) from board editor (i forced `e3` en passant flag)

it simply ignores it and forces `-` instead

**Comment by yo35** [comment]
> from game perspective - positions are equal

Agree. I'm not completely satisfied with the current behavior either.

Still, this en-passant flag is issue is a bit tricky, as checking for the presence of enemy paws in the squares close to the pawn that can potentially be captured en-passant is not enough. Consider for instance the following position:

![pos1](https://user-images.githubusercontent.com/5740068/183649415-53f4026e-e06e-4624-ad50-80f3963a5d84.png)

Even if it occurs after move c7-c5, en-passant is not possible because it would put the white king in check. Thus, both this position obtained after c7-c5 and this position obtained after a rook move for instance should be considered as equal - from game perspective.

Another example of such weird position with respect to en-passant flag sanitization:

![pos2](https://user-images.githubusercontent.com/5740068/183651475-10a604e8-ab0e-470b-9476-cf9fcfd13a55.png)

**Comment by mbutsykin** [comment]
Jeez :D
These are really tricky =/

**Comment by yo35** [comment]
The behavior or `Position.isEqual(..)` has been fixed in 3.1.0.
